test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/InstallCommandTests.cs

```diff
@@ -0,0 +1,254 @@
+            var nupkg = new SimpleTestPackageContext("A", "1.0.0");
+            var testServer = await _testFixture.GetSigningTestServerAsync();
+            var certificateAuthority = await _testFixture.GetDefaultTrustedCertificateAuthorityAsync();
+            var issueOptions = IssueCertificateOptions.CreateDefaultForEndCertificate();
+            var bcCertificate = certificateAuthority.IssueCertificate(issueOptions);
+
+            using (var context = new SimpleTestPathContext())
+            using (var testCertificate = new X509Certificate2(bcCertificate.GetEncoded()))
+            {
+                testCertificate.PrivateKey = DotNetUtilities.ToRSA(issueOptions.KeyPair.Private as RsaPrivateCrtKeyParameters);
+
+                var signedPackagePath = await SignedArchiveTestUtility.AuthorSignPackageAsync(testCertificate, nupkg, context.WorkingDirectory);
+                SignedArchiveTestUtility.TamperWithPackage(signedPackagePath);
+                certificateAuthority.Revoke(
+                    bcCertificate,
+                    RevocationReason.KeyCompromise,
+                    DateTimeOffset.UtcNow);
+
+                var args = new string[]
+                {
+                    nupkg.Id,
+                    "-Version",
+                    nupkg.Version,
+                    "-DirectDownload",
+                    "-NoCache",
+                    "-Source",
+                    context.WorkingDirectory,
+                    "-OutputDirectory",
+                    Path.Combine(context.WorkingDirectory, "packages")
+                };
+
+                // Act
+                var result = RunInstall(_nugetExePath, context, expectedExitCode: 1, additionalArgs: args);
+
+                // Assert
+                result.ExitCode.Should().Be(1);
+                result.Errors.Should().Contain(_NU3008);
+                result.Errors.Should().Contain(_NU3012);
+                result.AllOutput.Should().Contain($"WARNING: {_NU3027}");
+            }
+        }
+
+        public static CommandRunnerResult RunInstall(string nugetExe, SimpleTestPathContext pathContext, int expectedExitCode = 0, params string[] additionalArgs)
+        {
+            // Store the dg file for debugging
+            var envVars = new Dictionary<string, string>()
+            {
+                { "NUGET_HTTP_CACHE_PATH", pathContext.HttpCacheFolder }
+            };
+
+            var args = new string[] {
+                    "install",
+                    "-Verbosity",
+                    "detailed"
+                };
+
+            args = args.Concat(additionalArgs).ToArray();
+
+            // Act
+            var r = CommandRunner.Run(
+                nugetExe,
+                pathContext.WorkingDirectory,
+                string.Join(" ", args),
+                waitForExit: true,
+                environmentVariables: envVars);
+
+            // Assert
+            Assert.True(expectedExitCode == r.Item1, r.Item3 + "\n\n" + r.Item2);
```

Comment:
We have lots of existing code that uses `Item1`, `Item2`, and `Item3`, but they are terrible property names.  `CommandRunnerResult` already has highly readable properties which new code should prefer.

    Assert.True(expectedExitCode == r.ExitCode, r.AllOutput);